src/main/java/org/jitsi/eventadmin/EventAdminImpl.java

```diff
@@ -0,0 +1,235 @@
+ * instances. It has basic support for topic filtering by being able to process
+ * "*" wildcard.
+ *
+ * @author Pawel Domas
+ */
+public class EventAdminImpl
+    implements EventAdmin,
+               ServiceListener
+{
+    /**
+     * The logger.
+     */
+    static private final Logger logger = Logger.getLogger(EventAdminImpl.class);
+
+    /**
+     * The OSGi bundle context this instance is running on.
+     */
+    private BundleContext bundleContext;
+
+    /**
+     * Maps service reference to <tt>HandlerRef</tt> for convenience in
+     * add/remove operations.
+     */
+    private Map<ServiceReference<EventHandler>, HandlerRef> handlers
+        = new HashMap<>();
+
+    public void start(BundleContext ctx)
+        throws InvalidSyntaxException
+    {
+        ctx.addServiceListener(
+            this, "(objectclass=" + EventHandler.class.getName() + ")");
+
+        this.bundleContext = ctx;
+    }
+
+    public void stop(BundleContext ctx)
+    {
+        ctx.removeServiceListener(this);
+    }
+
+    /**
+     * Checks if <tt>EventHandler</tt> referenced in given <tt>HandlerRef</tt>
+     * is subscribed to given <tt>topic</tt>.
+     *
+     * @param handlerRef <tt>HandlerRef</tt> instance that refers to
+     *                   <tt>EventHandler</tt> which we're going to check.
+     * @param topic the event topics we're checking for.
+     *
+     * @return <tt>true</tt> if event handler is subscribed for given topic.
+     */
+    private boolean hasTopic(HandlerRef handlerRef, String topic)
+    {
+        return handlerRef.topicsPattern.matcher(topic).matches();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    synchronized public void sendEvent(Event event)
+    {
+        String eventTopic = event.getTopic();
+
+        if (StringUtils.isNullOrEmpty(eventTopic))
+        {
+            logger.warn(
+                "Some trash event without correct topic: " + eventTopic);
+            return;
+        }
+
+        for (HandlerRef handlerRef : handlers.values())
+        {
+            if (hasTopic(handlerRef, eventTopic))
+            {
+                try
+                {
+                    handlerRef.handler.handleEvent(event);
+                }
+                catch (Exception e)
+                {
+                    logger.error("EventHandler exception", e);
+                }
+            }
+        }
+    }
+
+    /**
+     * Creates {@link Pattern} that is supposed to match all the topics
+     * described by <tt>topic</tt> array. It does combine all the topics with
+     * 'or' regular expression. Also wildcard sign '*' is converted to Java
+     * regular expression equivalent '.*'.
+     *
+     * @param topics the array of event topics to be converted into
+     *               <tt>Pattern</tt>.
+     *
+     * @return <tt>Pattern</tt> that will match all the topics from
+     *         <tt>topic</tt> array.
+     */
+    Pattern createTopicsPattern(String[] topics)
```

Comment:
This be `static`, right? It doesn't make any functional difference at all, so you don't have to change this, but if you're skimming the code, it might help better understand what variables this method can potentially modify.